src/main/java/model/dataaccess/BrandDAO.java

```diff
@@ -60,17 +49,51 @@ public static void getConfig() {
      * @return Brand
      */
     public Brand getById(int id) {
-        Session session = factory.openSession();
         Brand brand = null;
-        Transaction tx = null;
         try {
-            tx = session.beginTransaction();
-            brand = session.load(Brand.class, id);
-        } catch (HibernateException e) {
-            if (tx != null) tx.rollback();
-            e.printStackTrace();
-        } finally {
-            session.close();
+            Session session = factory.openSession();
+            Transaction tx = null;
+            try {
+                tx = session.beginTransaction();
+                brand = session.load(Brand.class, id);
+            } catch (HibernateException e) {
+                if (tx != null) tx.rollback();
+                e.printStackTrace();
+            } finally {
+                session.close();
+            }
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+        }
+        return brand;
+    }
+
+    /**
+     * Method that returns a Brand based on a name
+     *
+     * @param name
+     * @return
+     */
+    @Override
+    public Brand getByName(String name) {
+        Brand brand = null;
+        try {
+            Session session = factory.openSession();
+            Transaction tx = null;
+            Criteria cr = null;
+            try {
+                tx = session.beginTransaction();
+                cr = session.createCriteria(Brand.class);
+                cr.add(Restrictions.eq("brandName", name));
+                brand = (Brand) cr.uniqueResult();

```

Comment:
brandName nu e setat ca si unique column in baza de date. 
UniqueResult pe brandName nu are cum sa functioneze fiindca e valid sa introduc doua branduri cu acelasi nume, inca.